ENT-6784: Change provisions for Artemis Server logging

vkolomeyko opened the pull request:
As a side effect of enabling warnings for Artemis was the following messages to be produced in the Node's log:
```
[WARN ] 2022-05-16T14:53:01,203Z [main] core.server. - AMQ222165: No Dead Letter Address configured for queue rpc.server in AddressSettings
[WARN ] 2022-05-16T14:53:01,203Z [main] core.server. - AMQ222166: No Expiry Address configured for queue rpc.server in AddressSettings
...
[WARN ] 2022-05-16T14:53:01,885Z [Thread-1 (ActiveMQ-server-org.apache.activemq.artemis.core.server.impl.ActiveMQServerImpl$6@70eb0c59)] core.server. - AMQ222165: No Dead Letter Address configured for queue p2p.inbound.DL5WQBEduTfiTzdLkg7iXcz5vY4oaJNdJr7Y4xaLvYszPj in AddressSettings
[WARN ] 2022-05-16T14:53:01,885Z [Thread-1 (ActiveMQ-server-org.apache.activemq.artemis.core.server.impl.ActiveMQServerImpl$6@70eb0c59)] core.server. - AMQ222166: No Expiry Address configured for queue p2p.inbound.DL5WQBEduTfiTzdLkg7iXcz5vY4oaJNdJr7Y4xaLvYszPj in AddressSettings
```

To suppress them a dedicated filter been added.
The change to logging configuration been tested locally using out-of-process integration tests.

rick-r3 commented:
Note, the following applies to Enterprise.  I'm not necessarily objecting to this OS PR.  We may or may not want to tackle differently between OS and ENT.

I would be very wary of putting any filters on the main "path" of log messages.  We've run into issues before where the filtering actually was the performance limiting factor of Corda nodes in the performance cluster.  Okay, in that instance it wasn't a regex but a javascript script (!!!), but still makes me wary. 

Is it possible to divert artemis messages to their own appender so only artemis messages are filtered?  I think the `Rewrite` appender functionality of log4j or some other feature might mean we can give just the artemis log messages to the filter by passing them through their own appender that then forwards on to the regular appender, as currently configured in the artemis section of that config.  There may be other ways to do that forwarding on to the original appender (and into the same log files).

vkolomeyko commented:
> Note, the following applies to Enterprise. I'm not necessarily objecting to this OS PR. We may or may not want to tackle differently between OS and ENT.
> 
> I would be very wary of putting any filters on the main "path" of log messages. We've run into issues before where the filtering actually was the performance limiting factor of Corda nodes in the performance cluster. Okay, in that instance it wasn't a regex but a javascript script (!!!), but still makes me wary.
> 
> Is it possible to divert artemis messages to their own appender so only artemis messages are filtered? I think the `Rewrite` appender functionality of log4j or some other feature might mean we can give just the artemis log messages to the filter by passing them through their own appender that then forwards on to the regular appender, as currently configured in the artemis section of that config. There may be other ways to do that forwarding on to the original appender (and into the same log files).

Thanks @rick-r3 - I can see your point. Let me try to make filtering more targeted at Artemis only.
(Also I think we should aim for aligning logging setup between OS and ENT as much as possible)

Commit:
ENT-6784: Change special provisions for Artemis Server logging

As a side effect of enabling warnings for Artemis was the following messages to be produced in the Node's log:
```
[WARN ] 2022-05-16T14:53:01,203Z [main] core.server. - AMQ222165: No Dead Letter Address configured for queue rpc.server in AddressSettings
[WARN ] 2022-05-16T14:53:01,203Z [main] core.server. - AMQ222166: No Expiry Address configured for queue rpc.server in AddressSettings
...
[WARN ] 2022-05-16T14:53:01,885Z [Thread-1 (ActiveMQ-server-org.apache.activemq.artemis.core.server.impl.ActiveMQServerImpl$6@70eb0c59)] core.server. - AMQ222165: No Dead Letter Address configured for queue p2p.inbound.DL5WQBEduTfiTzdLkg7iXcz5vY4oaJNdJr7Y4xaLvYszPj in AddressSettings
[WARN ] 2022-05-16T14:53:01,885Z [Thread-1 (ActiveMQ-server-org.apache.activemq.artemis.core.server.impl.ActiveMQServerImpl$6@70eb0c59)] core.server. - AMQ222166: No Expiry Address configured for queue p2p.inbound.DL5WQBEduTfiTzdLkg7iXcz5vY4oaJNdJr7Y4xaLvYszPj in AddressSettings
```

To suppress them a dedicated filter been added.
The change to logging configuration been tested locally using out-of-process integration tests.

rick-r3 commented on config/dev/log4j2.xml:
```diff
@@ -202,7 +202,11 @@
             <AppenderRef ref="RollingFile-ErrorCode-Appender"/>
         </Logger>
-        <Logger name="org.apache.activemq.artemis.core.server" level="error" additivity="false">
+        <Logger name="org.apache.activemq.artemis.core.server" level="info" additivity="false">
```
I was expecting "warn" but will defer to you if "info" is still not too bad verbosity-wise.

vkolomeyko replied:
Yes, `info` is not too bad verosity-wise. Also `info` might have some more useful information if Artemis will have a "hick-up" again.
